modules/microshift-oc-mirror-embed-ops-offline-use.adoc

```diff
@@ -0,0 +1,164 @@
+* You have a catalog index for the Operators you want to use.
+* jq is installed.
+* You have an Image Builder blueprint TOML file. If you have not created one, you can use the blueprint installed in the `/usr/share/microshift/blueprint` directory that is specific to your platform architecture.
+
+.Procedure
+
+. Parse the catalog index and get the image references by running the following example command:
++
+[source,terminal]
+----
+jq -r --slurp '.[] | select(.relatedImages != null) | "[[containers]]\nsource = \"" + .relatedImages[].image + "\"\n"'   ./oc-mirror-workspace/src/catalogs/registry.redhat.io/redhat/redhat-operator-index/v4.14/index/index.json
+----
+
+. Optional: Filter out images that cannot be mirrored by running the following example command:
++
+[source,terminal]
+----
+$ jq -r --slurp '.[] | select(.relatedImages != null) | .relatedImages[] | select(.name |  contains("ppc") or contains("s390x") | not) | "[[containers]]\\nsource = \\"" + .image + "\\"\\n"' ./oc-mirror-workspace/src/catalogs/registry.redhat.io/redhat/redhat-operator-index/v4.14/index/index.json
+----
++
+[NOTE]
+====
+This optional step uses the AMQ Broker Operator as an example. You can add other criteria to the `jq` command for further filtering as required by your use case.
+====
++
+.Example output
+[source,terminal]
+----
+[[containers]]
+source = "registry.redhat.io/amq7/amq-broker-init-rhel8@sha256:0b2126cfb6054fdf428c1f43b69e36e93a09a49ce15350e9273c98cc08c6598b"
+
+[[containers]]
+source = "registry.redhat.io/amq7/amq-broker-init-rhel8@sha256:0dde839c2dce7cb684094bf26523c8e16677de03149a0fff468b8c3f106e1f4f"
+...
+...
+
+[[containers]]
+source = "registry.redhat.io/amq7/amq-broker-rhel8@sha256:e8fa2a00e576ecb95561ffbdbf87b1c82d479c8791ab2c6ce741dd0d0b496d15"
+
+[[containers]]
+source = "registry.redhat.io/amq7/amq-broker-rhel8@sha256:ff6fefad518a6c997d4c5a6e475ba89640260167f0bc27715daf3cc30116fad1"
+…
+EOF
+----
+
+[IMPORTANT]
+====
+For mirrored and offline cases especially, ensure that all of the sources filtered from the `index.json` of your Operator are digests. Otherwise, the Operator installation fails.
+====
+
+. View the `imageset-config.yaml` to get the Operator image reference by running the following command:
++
+[source,terminal]
+----
+$ cat imageset-config.yaml
+----
++
+.Example output
+[source,terminal]
+----
+kind: ImageSetConfiguration
+apiVersion: mirror.openshift.io/v1alpha2
+storageConfig:
+  registry:
+    imageURL: registry.example.com/microshift-mirror
+mirror:
+  operators:
+  - catalog: registry.redhat.io/redhat/redhat-operator-index:v4.15 <1>
+    packages:
+    - name: amq-broker-rhel9
+      channels:
+      - name: 7.11.x
+----
+<1> Use the value in the `mirror.catalog` field as the catalog image reference for the blueprint.
+
+. Format the catalog image reference for the blueprint by following the example:
++
+[source,terminal]
+----
+[[containers]]
+source = "registry.redhat.io/redhat/redhat-operator-index:v4.15"
```

Comment:
Maybe we should use skopeo to get the digest at this moment? I think it would be clearer flow

Reply:
Would that look something like:

$ skopeo inspect --format "{{ .Digest }}" "docker://$image"

Reply:
Yeah, we have it later in the documentation and I was thinking if it would be clearer to get that digest earlier and only use it rather than embed using tag but then change to digest.